cbor/src/main/java/com/fasterxml/jackson/dataformat/cbor/CBORParser.java

```diff
@@ -2140,8 +2140,10 @@ protected void convertNumberToBigDecimal() throws IOException
         if ((_numTypesValid & (NR_DOUBLE | NR_FLOAT)) != 0) {
             // Let's parse from String representation, to avoid rounding errors that
             //non-decimal floating operations would incur
+            final String text = getText();
```

Comment:
Hmmmh. I am not sure the original logic makes much sense. We must have decoded underlying FP and seems like going into String-representation does nothing to improve or maintain accuracy... 

But... let me actually add comments at higher level as main thread, wrt limits

Reply:
this code already uses getText() - so I'm just applying the string length check here

There is also unmodified code for handling binary encoded numbers. I haven't changed that code yet and it would be good to do this too - but it isn't as obvious how to check the len in that flow.

An example from the Smile parsing code:
```
        if (_decode7BitEncoded()) { // got it all!
            // note: scale value is signed, needs zigzag, so:
            int scale = SmileUtil.zigzagDecode((int) _pending64);
            BigInteger bigInt = new BigInteger(_byteArrayBuilder.toByteArray());
            _numberBigDecimal = new BigDecimal(bigInt, scale);
            _numberType = NumberType.BIG_DECIMAL;
            _numTypesValid = NR_BIGDECIMAL;
            return _valueComplete(JsonToken.VALUE_NUMBER_FLOAT);
        }
```

So we could check the len of the _byteArrayBuilder.toByteArray() and fail if it is too long.

Reply:
@pjfanning Right, yes, it is existing handling, did not think you were changing. So more a note to myself. :)